### PR TITLE
libusermode: DLL search case insensitive, simple wildcard pattern

### PR DESCRIPTION
**Pull request by skvl**
Simplifies rules for use in `apimon` while DLL name comparison:
* Allow simple wildcard patter `*.dll`.
* Performs case insensitive search.

### CODE REVIEW

**Comment by skvl** [comment]
@tklengyel , I believe that astyle check is bad for proposed lambda expression.

**Comment by skvl** [comment]
> Feel free to tweak the .astylerc

Astyle has problems with lamdas:
* https://sourceforge.net/p/astyle/tickets/121/
* https://sourceforge.net/p/astyle/bugs/317/
* https://sourceforge.net/p/astyle/bugs/448/

So i don't know what to do here.

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

File: src/libusermode/userhook.cpp [hunk 1]
```diff
@@ -1073,3 +1073,20 @@
 {
     return userhook::is_supported(drakvuf);
 }
+
+static bool dll_name_comparator(char x, char y)
```
[the remark this answers is not in the document]
Comment (skvl) [reply src/libusermode/userhook.cpp:1077]: This is not the issue. The function pointer is used below.

**Comment by skvl** [comment]
@tklengyel , is this OK?

**Pull request merged**